tests/federation/test_federation_server.py

```diff
@@ -84,6 +86,55 @@ async def failing_handler(_origin: str, _content: JsonDict) -> None:
         )
         self.assertEqual(500, channel.code, channel.result)
 
+    def test_accept_valid_pdus_and_ignore_invalid(self) -> None:
+        user = self.register_user("nex", "test")
+        tok = self.login("nex", "test")
+        room_id = self.helper.create_room_as("nex", tok=tok)
+
+        builder = self.hs.get_event_builder_factory().for_room_version(
+            RoomVersions.V10,
+            {
+                "type": EventTypes.Message,
+                "sender": user,
+                "room_id": room_id,
+                "content": {"body": "hello i am nexy", "msgtype": "m.text"},
```

Comment:
Would be good to label them for easier debugging

```suggestion
                "content": {"body": "event1", "msgtype": "m.text"},
```